### PR TITLE
[BE][TP] Check module has bias before access

### PR DESCRIPTION
**Pull request by kwen2501**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132137

Some linear modules, such as the ones reconstructed by `torch.export.unflatten()`, may not have the `bias` attribute, if the original linear module has `bias=None`.

cc @XilunWu @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

File: torch/distributed/tensor/parallel/style.py [hunk 1]
```diff
@@ -227,7 +227,7 @@ def _partition_linear_fn(self, name, module, device_mesh):
             "weight",
             nn.Parameter(distribute_tensor(module.weight, device_mesh, [Shard(1)])),
         )
-        if module.bias is not None:
+        if hasattr(module, "bias") and module.bias is not None:
```
Comment (awgu) [inline torch/distributed/tensor/parallel/style.py:230]:
```suggestion
        if getattr(module, "bias", None) is not None:
```

**Review by wanchaol — APPROVED** [review]

File: torch/distributed/tensor/parallel/style.py (hunk 1 above)
Comment (kwen2501) [reply torch/distributed/tensor/parallel/style.py]: Thx! Adopted!

**Pull request closed without merging**